src/Reopt.hs

```diff
@@ -2694,20 +2699,30 @@ reoptRecoveryLoop symAddrMap rOpts funPrefix sysp debugTypeMap = go
     let unexplored = newDiscState ^. Macaw.unexploredFunctions
 
     if null unexplored
-      then traceM "NOLOOP" >> return (newDiscState, recoverX86Output, recMod, moduleConstraints)
-      else traceM "LOOP" >> go newDiscState
+      then return (newDiscState, recoverX86Output, recMod, moduleConstraints)
+      else go newDiscState
 
+newtype IntendToRecover = IntendToRecover {getIntendToRecover :: Bool}
```

Comment:
Mention that this type is only used as an argument to `reoptPrepareForRecovery`.

Reply:
You know what, I don't know why this needs to happen so early, since it's about doing checks if you're trying to do recovery. I'm going to see whether we can just do these checks as we're about to start recovery, seems less silly.

Reply:
I removed this flag entirely, and made it so that the check happens just as recovery is about to happen, seems better to me. Also renamed "prepare for recovery" into "initial discovery" as this describes better what it does.